Makefile

```diff
@@ -52,19 +52,15 @@ install-lll:
 check-fmt: install-lll
 	etc/check_fmt.sh
 
-# check-modules runs "go mod tidy" then "go mod vendor" and exits with a non-zero exit code if there
-# are any module or vendored modules changes. The intent is to confirm two properties:
-#
-# 1. Exactly the required modules are declared as dependencies. We should always be able to run
-# "go mod tidy" and expect that no unrelated changes are made to the "go.mod" file.
-#
-# 2. All required modules are copied into the vendor/ directory and are an exact copy of the
-# original module source code (i.e. the vendored modules are not modified from their original code).
+# check-modules runs "go mod tidy"  and exits with a non-zero exit code if there
+# are any module changes. The intent is to confirm that exactly the required 
+# modules are declared as dependencies. We should always be able to run "go mod 
+# tidy" and expect that no unrelated changes are made to the "go.mod" file.
 .PHONY: check-modules
 check-modules:
 	go mod tidy -v
-	go mod vendor
-	git diff --exit-code go.mod go.sum ./vendor
+	go mod verify
```

Comment:
Optional: `go mod verify` seems to conflict with using Go workspaces in some cases (see https://github.com/golang/go/issues/62663). When I run that command on my workstation, I get an error:
```
❯ go mod verify
go.mongodb.org/mongo-driver : missing ziphash: open hash: no such file or directory
```
Consider removing `go mod verify`.